src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/EventPipeSession.cs

```diff
@@ -30,16 +29,84 @@ private EventPipeSession(IpcEndpoint endpoint, IpcResponse response, ulong sessi
 
         internal static EventPipeSession Start(IpcEndpoint endpoint, EventPipeSessionConfiguration config)
         {
-            IpcMessage requestMessage = CreateStartMessage(config);
+            IpcMessage requestMessage = CreateStartMessage(config, EventPipeCommandId.CollectTracing4);
             IpcResponse? response = IpcClient.SendMessageGetContinuation(endpoint, requestMessage);
-            return CreateSessionFromResponse(endpoint, ref response, nameof(Start));
+            try
+            {
+                EventPipeSession session = null;
+                if (DiagnosticsClient.ValidateResponseMessage(response.Value.Message, nameof(StartAsync), DiagnosticsClient.ValidateResponseOptions.UnknownCommandReturnsFalse))
+                {
+                    session = CreateSessionFromResponse(endpoint, ref response);
+                    // The session owns the continuation stream
+                    response = null;
+                    return session;
+                }
+
+                requestMessage = CreateStartMessage(config, EventPipeCommandId.CollectTracing3);
```

Comment:
Falling back to `CollectTracing3` may not be the right thing to do if not having the custom keyword is fatal. (There is no such scenario right now, but I am not sure how can this model support that scenario if we have that case in the future.

And you might not need `CollectTracing3` if you don't need to exclude stacks here.